modules/flatpak/install.nix

```
@@ -186,48 +184,107 @@ let
 
   flatpakInstall = installation: update: packages: map (flatpakInstallCmd installation update) packages;
 
-  mkFlatpakInstallCmd = installation: update: packages: builtins.foldl' (x: y: x + y) '''' (flatpakInstall installation update packages);
 
   flatpakDeleteRemotesCmd = remotes.flatpakDeleteRemotesCmd;
-  mkFlatpakAddRemotesCmd = remotes.mkFlatpakAddRemotesCmd;
-in
-pkgs.writeShellScript "flatpak-managed-install" ''
-  # This script is triggered at build time by a transient systemd unit.
-  set -eu
-
-  # Setup state variables for packages and remotes
-  NEW_STATE=$(${pkgs.coreutils}/bin/cat ${stateFile})
-  if [[ -f ${statePath} ]]; then
-    OLD_STATE=$(${pkgs.coreutils}/bin/cat ${statePath})
-  else
-    OLD_STATE={}
-  fi
-
-  # Handle unmanaged packages and remotes.
-  ${handleUnmanagedStateCmd installation cfg.uninstallUnmanaged}
-
-  # Configure remotes
-  ${mkFlatpakAddRemotesCmd installation cfg.remotes}
-
-  # Uninstall packages that have been removed from services.flatpak.packages
-  # since the previous activation.
-  ${flatpakUninstallCmd installation {}}
 
-  # Uninstall remotes that have been removed from services.flatpak.packages
-  # since the previous activation.
-  ${flatpakDeleteRemotesCmd installation cfg.uninstallUnmanaged {}}  
-
-  # Install packages
-  ${mkFlatpakInstallCmd installation updateApplications cfg.packages}
-
-  # Configure overrides
-  ${flatpakOverridesCmd installation {}}
+  updateTrigger =
+    if executionContext == "service-start" then cfg.update.onActivation
+    else if executionContext == "timer" then cfg.update.auto.enable
+    else lib.warn ("executionContext=${executionContext}: invalid arugment.") false;
```

here we should fail, not set  to `false`